### PR TITLE
Update LinchPin Merge trigger to perform release

### PR DESCRIPTION
**Pull request by herlo**
When a new version of LinchPin is pushed and passes tests. The [merge] comment can be issued, which will first merge the PR. Then, the job will check to see if a new `linchpin/version.py` was created. If so, it will trigger a release. If the release already exists, the cd-linchpin-release will fail, resulting in only *new* releases being pushed.

### CODE REVIEW

**Comment by herlo** [comment]
[test]

**Comment by herlo** [comment]
[test]

**Comment by herlo** [comment]
Was experiencing a weird git pull error for downstream tests. When run manually, however, things pass. See https://jenkins-continuous-infra.cloud.continuous-infra.upshift.redhat.com/job|
/ci-linchpin/289/console

**Comment by herlo** [comment]
@p3ck Another review?

File: config/Dockerfiles/JenkinsfileGHPRMerge [hunk 1]
```diff
@@ -100,5 +100,58 @@ pipeline {
...
+
+
+                                // make sure to put this to == before merging
+                                // the regex checks if __version__ = '1.6.2a1' or the like.
+                                // if __version__ matches this, do nothing
+                                // if __version__ returns blank, we have good confidence
+                                // it's a full release version.
+                                // cd-linchpin-release will attempt a release
+                                if (version == "") {
+                                    println "release version == ${version}"
```
[the remark this answers is not in the document]
Comment (herlo) [reply config/Dockerfiles/JenkinsfileGHPRMerge]: @p3ck this is the logic error, I just pushed an updated to fix this.

[the remark this answers is not in the document]
Comment (p3ck) [reply config/Dockerfiles/JenkinsfileGHPRMerge]: This still looks wrong to me.  and I still think the name of the variable could be better.
Also - Please look at LinchpinPipelineUtils.groovy in linchpin/src/org/centos/pipeline.  That should show you how to look at the changed files in this PR without having to call out to a script all the time.

File: config/Dockerfiles/JenkinsfileGHPRMerge [hunk 2]
```diff
@@ -100,5 +100,56 @@ pipeline {
...
+                                    returnStdout: true).trim()
+
+
+                                // make sure to put this to == before merging
+                                // the regex checks if __version__ = '1.6.2a1' or the like.
+                                // if __version__ matches this, do nothing
+                                // if __version__ returns blank, we have good confidence
+                                // it's a full release version.
+                                // cd-linchpin-release will attempt a release
+                                if (pre_release == "") {
```
Comment (herlo) [inline config/Dockerfiles/JenkinsfileGHPRMerge]: @p3ck This should fix that concern.

**Comment by herlo** [comment]
[merge]

**Comment by herlo** [comment]
@p3ck I'll add an issue to make sure we get the code refactored. Good plan!

**Pull request merged**